staging/src/k8s.io/kubectl/pkg/cmd/logs/logs.go

```diff
@@ -319,6 +319,10 @@ func (o LogsOptions) RunLogs() error {
 		return err
 	}
 
+	if len(requests) == 0 {
```

Comment:
There is no point of running the validation of the options if there are no resources found this should be bailed out earlier in Complete function itself, without the needing to go through the Validate() and RunLogs() function

-srini

Reply:
@bseenu
Complete function is for checking required arguments is valid and Validate function is for checking optional flags is valid. These have to precede before the client sends a request.
(I think the logic of retrieving the pod list in Complete function also has to moves to RunLog function.)
Before the client sends a request, it's impossible to check resources exist or not. So it should be in RunLogs function.

Reply:
@cprayer i am still not convinced about validating the log options if there are no resources for logs

I would wait to see what others think about this?

Reply:
@bseenu, I don’t understand, can you explain your thinking further?  

I think the args are valid at this point, it is just that there could be nothing found in the cluster and this check appears to handle that by returning an error instead of succeeding silently.

It seems ok to me, but maybe I am missing something.

Reply:
@cprayer I have checked for other commands and see that resourcelist is built in the run function only, Not sure if there is any reasoning for this command to be built in the Complete function ( anyone? ), you can move the resource builder to RunLogs function , if it has to be in Complete function then we can error out there itself

-srini

Reply:
I agree that the solution proposed in https://github.com/kubernetes/kubernetes/pull/89815 seems better but this one adds a test, which is very important to have. @cprayer @bseenu try to figure out and combine these two PRs together.